src/domain/synthetics/positions/utils.ts

```diff
@@ -211,16 +212,36 @@ export function formatLiquidationPrice(liquidationPrice?: bigint, opts: { displa
   if (liquidationPrice === undefined || liquidationPrice < 0) {
     return "NA";
   }
+  const priceDecimalPlaces = calculatePriceDecimals(liquidationPrice);
 
-  return formatUsd(liquidationPrice, { ...opts, maxThreshold: "1000000" });
+  return formatUsd(liquidationPrice, {
+    ...opts,
+    displayDecimals: opts.displayDecimals ?? priceDecimalPlaces,
+    maxThreshold: "1000000",
+  });
 }
 
 export function formatAcceptablePrice(acceptablePrice?: bigint, opts: { displayDecimals?: number } = {}) {
   if (acceptablePrice !== undefined && (acceptablePrice == 0n || acceptablePrice >= ethers.MaxInt256)) {
     return "NA";
   }
 
-  return formatUsd(acceptablePrice, { ...opts });
+  const priceDecimalPlaces = calculatePriceDecimals(acceptablePrice);
+
+  return formatUsd(acceptablePrice, { ...opts, displayDecimals: opts.displayDecimals ?? priceDecimalPlaces });
+}
+
+export function formatUsdPrice(price?: bigint, opts: { displayDecimals?: number } = {}) {
```

Comment:
let's move this function to lib/numbers.ts

Reply:
you actually don't use displayDecimals and this param a bit contradicts the purpose of the function, let's stick to KISS approach here?

Reply:
agree on both